pkgs/os-specific/linux/cryptsetup/default.nix

```diff
@@ -38,6 +38,8 @@ stdenv.mkDerivation rec {
   buildInputs = [ lvm2 json_c openssl libuuid popt ]
     ++ stdenv.lib.optional enablePython python2;
 
+  doCheck = true;
```

Comment:
What was the failure message?

Reply:
Ideally we won't stuck on older versions forever or else we can get security updates.

Reply:
It fails most "compat" tests with some strange "file not found" errors. It might be just a bug in the tests, but I would feel uneasy running `cryptsetup` that could break something that is "compat".

Reply:
Is this with or without sandbox? The latest cryptsetup also checks block devices for other filesystems and prevent corruption that way. That's means I would also feel uneasy to not having this safety feature.

Reply:
I cannot reproduce the failure. Tests works for me: https://gist.github.com/Mic92/a603aea57ecd079b76903a54d3e1dd90#file-cryptsetup-log-L406

Reply:
With sandbox. I get this https://gist.github.com/oxij/460d00b1099768f0641b1c3b6dac5791 failure on every rebuilt try. Kernel 4.9, building on tmpfs.

Reply:
Your log has
> Nfs? I'm not going to run blockwise tests on this.
so this test is skipped on your machine for some reason.

Reply:
Same failure as for @oxij here. Kernel 4.14, sandbox,  master 7bcc2870c75

Reply:
I assume it detects my filesystem as `nfs` because I use zfs. Do we really want tests that depend on the build machines filesystem? IMHO those should better run in a VM.

Reply:
The works for on 4.4 with `ext4` without sandbox.

Reply:
ok, File not found could be the script itself, please try to use `patchShebangs` instead:

https://gitlab.com/cryptsetup/cryptsetup/blob/master/tests/compat-test#L1

Reply:
There seem to be 2 issues here: 

1. The `file not found` error above is sandbox-related (although patchShebangs is already used). Still trying to find the reason.

2. With sandboxing disabled, the `compat` test runs and succeeds if the build directory is on an ext4 fs, but 2 test cases fail if the build directory is on a btrfs fs (which is my default here). This is weird. We cannot have test success depend on the what fs the build dir is on, so we will need to disable the failing cases.

Reply:
No, actually it's only one issue: the test uses an internal tool `unit-utils-io` to read/write blocks in the file `blocwise_localfile`. This file exists but `unit-utils-io` opens it with `O_DIRECT` https://gitlab.com/cryptsetup/cryptsetup/blob/master/tests/unit-utils-io.c#L92, which is forbidden on a tmpfs and on btrfs (which produces the `file not found` errors - tmpfs in sandbox, btrfs without sandbox) but allowed on ext4. This is just stupid.

@oxij  @Mic92 what should we do? We could 

- revert to 2.0.3 
- just disable this filesystem-dependent test
- create a nixos test to run complete the tests in a VM where we can control the file system type.

Reply:
Either 

- disable this filesystem-dependent test
- create a nixos test to run complete the tests in a VM where we can control the file system type.

O_DIRECT is not implemented on zfs. It is a braindead interface anyway.
Reverting just because a test does not work in our build environment is pointless as the test works perfectly fine without it.
Actually we already have integration tests for cryptsetup. And don't think we need to duplicate upstream testing beyond a certain level in our own systems. Most of the critical stuff in cryptsetup is implemented in the kernel anyway.

Reply:
I'm implementing a solution to patch out O_DIRECT and disable the remaining 4 test cases. PR will follow shortly.